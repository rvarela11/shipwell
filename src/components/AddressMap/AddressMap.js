/* eslint-disable camelcase */
/* global google */

// @vendors
import React from 'react';
import { compose, withProps, lifecycle } from 'recompose';
import {
    withScriptjs,
    withGoogleMap,
    GoogleMap,
    DirectionsRenderer
} from 'react-google-maps';
import { connect } from 'react-redux';

export const AddressMap = compose(
    withProps({
        googleMapURL: 'https://maps.googleapis.com/maps/api/js?key=AIzaSyCtJUNx9yMHeFXIFA2BWIX8z8t007rCpU0&v=3.exp&libraries=geometry,drawing,places',
        loadingElement: <div style={{ height: '400px' }} />,
        containerElement: <div style={{ width: '100%' }} />,
        mapElement: <div style={{ height: '50vh', width: '100%' }} />
    }),
    withScriptjs,
    withGoogleMap,
    lifecycle({
        componentDidMount() {
            const {
                latitude_address_Origin,
                longitude_address_Origin,
                latitude_address_Destination,
                longitude_address_Destination
            } = this.props;
            const directionsService = new google.maps.DirectionsService();
            directionsService.route({
                origin: new google.maps.LatLng(latitude_address_Origin, longitude_address_Origin),
                destination: new google.maps.LatLng(latitude_address_Destination, longitude_address_Destination),
                travelMode: google.maps.TravelMode.DRIVING
            }, (response, status) => {
                if (status === google.maps.DirectionsStatus.OK) {
                    this.setState({
                        directions: { ...response },
                        markers: true
                    });
                } else {
                    console.error(`error fetching directions ${response}`);
                }
            });
        }
    })
)((props) => {
    const { directions, markers } = props;
    return (
        <GoogleMap defaultZoom={3}>
            {directions && <DirectionsRenderer directions={directions} suppressMarkers={markers} />}
        </GoogleMap>
    );
});

const mapStateToProps = state => ({
    latitude_address_Origin: state.latitude_address_Origin,
    longitude_address_Origin: state.longitude_address_Origin,
    latitude_address_Destination: state.latitude_address_Destination,
    longitude_address_Destination: state.longitude_address_Destination
});

export default connect(mapStateToProps, null)(AddressMap);
