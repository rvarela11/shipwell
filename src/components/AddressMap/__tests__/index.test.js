// @vendors
import React from 'react';
import { shallow } from 'enzyme';

// @components
import { AddressMap } from '../AddressMap';

describe('AddressMap', () => {
    it('renders AddressMap without crashing', () => {
        const wrapper = shallow(
            <AddressMap />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
