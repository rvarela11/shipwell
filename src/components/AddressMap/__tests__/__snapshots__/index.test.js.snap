// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressMap renders AddressMap without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withProps(withScriptjs(withGoogleMap(lifecycle(Component)))) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "containerElement": <div
        style={
          Object {
            "width": "100%",
          }
        }
      />,
      "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyCtJUNx9yMHeFXIFA2BWIX8z8t007rCpU0&v=3.exp&libraries=geometry,drawing,places",
      "loadingElement": <div
        style={
          Object {
            "height": "400px",
          }
        }
      />,
      "mapElement": <div
        style={
          Object {
            "height": "50vh",
            "width": "100%",
          }
        }
      />,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "containerElement": <div
          style={
            Object {
              "width": "100%",
            }
          }
        />,
        "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyCtJUNx9yMHeFXIFA2BWIX8z8t007rCpU0&v=3.exp&libraries=geometry,drawing,places",
        "loadingElement": <div
          style={
            Object {
              "height": "400px",
            }
          }
        />,
        "mapElement": <div
          style={
            Object {
              "height": "50vh",
              "width": "100%",
            }
          }
        />,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
